pkg/reconciler/subscription/subscription_test.go

```diff
@@ -0,0 +1,700 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package subscription
+
+import (
+	"encoding/json"
+	"fmt"
+	testing2 "github.com/knative/eventing/pkg/reconciler/testing"
+	"testing"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/sets"
+	fakekubeclientset "k8s.io/client-go/kubernetes/fake"
+	"k8s.io/client-go/kubernetes/scheme"
+	clientgotesting "k8s.io/client-go/testing"
+
+	"github.com/knative/eventing/pkg/apis/duck/v1alpha1"
+	eventingv1alpha1 "github.com/knative/eventing/pkg/apis/eventing/v1alpha1"
+	fakeclientset "github.com/knative/eventing/pkg/client/clientset/versioned/fake"
+	informers "github.com/knative/eventing/pkg/client/informers/externalversions"
+	"github.com/knative/eventing/pkg/reconciler"
+	"github.com/knative/eventing/pkg/utils"
+	duckv1alpha1 "github.com/knative/pkg/apis/duck/v1alpha1"
+	"github.com/knative/pkg/controller"
+	logtesting "github.com/knative/pkg/logging/testing"
+	. "github.com/knative/pkg/reconciler/testing"
+)
+
+const (
+	subscriberName = "subscriber"
+	replyName      = "reply"
+	channelName    = "origin"
+	serviceName    = "service"
+
+	subscriptionUID  = subscriptionName + "-abc-123"
+	subscriptionName = "testsubscription"
+	testNS           = "testnamespace"
+)
+
+// subscriptions have: channel -> SUB -> subscriber -viaSub-> reply
+
+var (
+	channelDNS = "channel.mynamespace.svc." + utils.GetClusterDomainName()
+	channelURI = "http://" + channelDNS + "/"
+
+	subscriberDNS = "subscriber.mynamespace.svc." + utils.GetClusterDomainName()
+	subscriberURI = "http://" + subscriberDNS + "/"
+
+	replyDNS = "reply.mynamespace.svc." + utils.GetClusterDomainName()
+	replyURI = "http://" + replyDNS + "/"
+
+	serviceDNS = serviceName + "." + testNS + ".svc." + utils.GetClusterDomainName()
+	serviceURI = "http://" + serviceDNS + "/"
+
+	subscriberGVK = metav1.GroupVersionKind{
+		Group:   "testing.eventing.knative.dev",
+		Version: "v1alpha1",
+		Kind:    "Subscriber",
+	}
+
+	serviceGVK = metav1.GroupVersionKind{
+		Version: "v1",
+		Kind:    "Service",
+	}
+
+	channelGVK = metav1.GroupVersionKind{
+		Group:   "eventing.knative.dev",
+		Version: "v1alpha1",
+		Kind:    "Channel",
+	}
+)
+
+func init() {
+	// Add types to scheme
+	_ = eventingv1alpha1.AddToScheme(scheme.Scheme)
+	_ = duckv1alpha1.AddToScheme(scheme.Scheme)
+}
+
+func TestAllCases(t *testing.T) {
+	table := TableTest{
+		{
+			Name: "bad workqueue key",
+			// Make sure Reconcile handles bad keys.
+			Key: "too/many/parts",
+		}, {
+			Name: "key not found",
+			// Make sure Reconcile handles good keys that don't exist.
+			Key: "foo/not-found",
+			//}, { // TODO: there is a bug in the controller, it will query for ""
```

Comment:
Is this a new test you wrote @n3wscott? Good catch!

Reply:
no, it is an existing test that was blind to what the reconciler was really doing.